### PR TITLE
Coverage and completeness over exons of a sample endpoint

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- Adds and endpoint that accepts a sample name and gene list and returns coverage and completeness over the exons of these genes

### How to test:
- Launch an instance of the app and test the endpoint 

### Expected outcome:
- Exon coverage and completeness values should be returned

### Review:
- [ ] Code approved by
- [x] Tests executed by CR

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by northwestwitch** [comment]
Running the following query:

image


Returns:

```
[
  {
    "chromosome": "1",
    "completeness": [
      [
        10,
        0.9903881700554529
      ],
      [
        20,
        0.7323475046210721
      ],
      [
        30,
        0.06580406654343808
      ]
    ],
    "ensembl_gene_id": "ENSG00000143632",
    "end": 229569845,
    "hgnc_id": 129,
    "hgnc_symbol": "ACTA1",
    "interval_id": null,
    "mean_coverage": 42.78251999336164,
    "start": 229566992
  },
  {
    "chromosome": "18",
    "completeness": [
      [
        10,
        0.9944753145557897
      ],
      [
        20,
        0.6350514147026889
      ],
      [
        30,
        0.1607907006450789
      ]
    ],
    "ensembl_gene_id": "ENSG00000101680",
    "end": 7117813,
    "hgnc_id": 6481,
    "hgnc_symbol": "LAMA1",
    "interval_id": null,
    "mean_coverage": 222.1742068291945,
    "start": 6941743
  },
  {
    "chromosome": "7",
    "completeness": [
      [
        10,
        0.9790655204667944
      ],
      [
        20,
        0.8557302570041424
      ],
      [
        30,
        0.28217005924012295
      ]
    ],
    "ensembl_gene_id": "ENSG00000128513",
    "end": 124570037,
    "hgnc_id": 17284,
    "hgnc_symbol": "POT1",
    "interval_id": null,
    "mean_coverage": 269.20873768354465,
    "start": 124462440
  }
]
```

File: src/chanjo2/endpoints/coverage.py [hunk 1]
```diff
@@ -87,13 +90,10 @@ def d4_intervals_coverage(coverage_file_path: str, bed_file: bytes = File(...)):
...
 
 
-@router.post("/coverage/sample/genes_coverage", response_model=List[CoverageInterval])
-async def sample_genes_coverage(
-    query: SampleGeneQuery, db: Session = Depends(get_session)
-):
-    """Returns coverage over a list of genes (entire gene) for a given sample in the database."""
-
-    sample: SQLSample = get_sample(db=db, sample_name=query.sample_name)
+def get_sample_coverage_file(
```
Comment (northwestwitch) [inline src/chanjo2/endpoints/coverage.py:93]: A new function that replace repeated code in the 3 endpoints sample_genes_coverage, sample_transcripts_coverage and sample_esons_coverage

File: src/chanjo2/meta/handle_d4.py [hunk 2]
```diff
@@ -120,31 +120,32 @@ def get_genes_coverage_completeness(
     return genes_cov
 
 
-def get_transcripts_coverage_completeness(
+def get_gene_interval_coverage_completeness(
```
Comment (northwestwitch) [inline src/chanjo2/meta/handle_d4.py:123]: Modified this function to be invoked by both sample_transcripts_coverage and sample_econs_coverage

**Review by henrikstranneheim — APPROVED** [review]
Looks great ⭐

**Pull request merged**